[2024-06-30T09:01:47.506+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-30T09:01:47.527+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: northwindDWH.data_delivery manual__2024-06-30T08:53:53.031867+00:00 [queued]>
[2024-06-30T09:01:47.535+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: northwindDWH.data_delivery manual__2024-06-30T08:53:53.031867+00:00 [queued]>
[2024-06-30T09:01:47.535+0000] {taskinstance.py:2306} INFO - Starting attempt 2 of 2
[2024-06-30T09:01:47.546+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): data_delivery> on 2024-06-30 08:53:53.031867+00:00
[2024-06-30T09:01:47.555+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=162) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-06-30T09:01:47.556+0000] {standard_task_runner.py:63} INFO - Started process 164 to run task
[2024-06-30T09:01:47.556+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'northwindDWH', 'data_delivery', 'manual__2024-06-30T08:53:53.031867+00:00', '--job-id', '97', '--raw', '--subdir', 'DAGS_FOLDER/northwindDWH.py', '--cfg-path', '/tmp/tmpkkwyb2hj']
[2024-06-30T09:01:47.558+0000] {standard_task_runner.py:91} INFO - Job 97: Subtask data_delivery
[2024-06-30T09:01:47.569+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/settings.py:195: DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
  SQL_ALCHEMY_CONN = conf.get("database", "SQL_ALCHEMY_CONN")

[2024-06-30T09:01:47.595+0000] {task_command.py:426} INFO - Running <TaskInstance: northwindDWH.data_delivery manual__2024-06-30T08:53:53.031867+00:00 [running]> on host 3abb3aef8f21
[2024-06-30T09:01:47.679+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Mahmoud' AIRFLOW_CTX_DAG_ID='northwindDWH' AIRFLOW_CTX_TASK_ID='data_delivery' AIRFLOW_CTX_EXECUTION_DATE='2024-06-30T08:53:53.031867+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-06-30T08:53:53.031867+00:00'
[2024-06-30T09:01:47.679+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-30T09:01:47.692+0000] {northwindDWH.py:49} INFO - Data delivery started
[2024-06-30T09:01:47.692+0000] {Load_data.py:85} INFO - loading dimEmployee to db ...
[2024-06-30T09:01:47.692+0000] {Transform_data.py:51} INFO - start Extracting dim employee
[2024-06-30T09:01:47.695+0000] {db_utils.py:21} INFO - Connected to PostgreSQL database successfully
[2024-06-30T09:01:47.699+0000] {Transform_data.py:60} INFO - Extract dim employees done
[2024-06-30T09:01:47.699+0000] {db_utils.py:30} INFO - PostgreSQL connection is closed
[2024-06-30T09:01:47.703+0000] {db_utils.py:21} INFO - Connected to PostgreSQL database successfully
[2024-06-30T09:01:47.705+0000] {Load_data.py:103} INFO - SCD type 2 on dimemployee started
[2024-06-30T09:01:47.711+0000] {db_utils.py:30} INFO - PostgreSQL connection is closed
[2024-06-30T09:01:47.711+0000] {Load_data.py:159} INFO - Loading dimproduct to db...
[2024-06-30T09:01:47.712+0000] {Transform_data.py:27} INFO - Start Extracting dim product
[2024-06-30T09:01:47.715+0000] {db_utils.py:21} INFO - Connected to PostgreSQL database successfully
[2024-06-30T09:01:47.717+0000] {Transform_data.py:41} INFO - Extract dim products done
[2024-06-30T09:01:47.718+0000] {db_utils.py:30} INFO - PostgreSQL connection is closed
[2024-06-30T09:01:47.721+0000] {db_utils.py:21} INFO - Connected to PostgreSQL database successfully
[2024-06-30T09:01:47.723+0000] {Load_data.py:172} INFO - SCD type 1 on dimproduct started
[2024-06-30T09:01:47.724+0000] {Load_data.py:227} ERROR - Error while loading dimProducts: 'Product_id'
[2024-06-30T09:01:47.724+0000] {Load_data.py:12} INFO - loading dimCustomers to db ...
[2024-06-30T09:01:47.724+0000] {Transform_data.py:7} INFO - start Extracting dim customers
[2024-06-30T09:01:47.727+0000] {db_utils.py:21} INFO - Connected to PostgreSQL database successfully
[2024-06-30T09:01:47.729+0000] {Transform_data.py:16} INFO - Extract dim customer done
[2024-06-30T09:01:47.729+0000] {db_utils.py:30} INFO - PostgreSQL connection is closed
[2024-06-30T09:01:47.732+0000] {db_utils.py:21} INFO - Connected to PostgreSQL database successfully
[2024-06-30T09:01:47.734+0000] {Load_data.py:30} INFO - SCD type 2 on dimcustomer started
[2024-06-30T09:01:47.747+0000] {db_utils.py:30} INFO - PostgreSQL connection is closed
[2024-06-30T09:01:47.748+0000] {Load_data.py:232} INFO - Loading factsales to db ...
[2024-06-30T09:01:47.748+0000] {Transform_data.py:70} INFO - start Extracting fact sales
[2024-06-30T09:01:47.751+0000] {db_utils.py:21} INFO - Connected to PostgreSQL database successfully
[2024-06-30T09:01:47.754+0000] {Transform_data.py:81} INFO - Extract fact sales done
[2024-06-30T09:01:47.755+0000] {db_utils.py:30} INFO - PostgreSQL connection is closed
[2024-06-30T09:01:47.757+0000] {Tables_creations.py:27} INFO - Loading data into table fact_sales...
[2024-06-30T09:01:47.760+0000] {db_utils.py:21} INFO - Connected to PostgreSQL database successfully
[2024-06-30T09:01:47.777+0000] {Tables_creations.py:41} INFO - Data loaded successfully into table fact_sales With 58 Row
[2024-06-30T09:01:47.778+0000] {db_utils.py:30} INFO - PostgreSQL connection is closed
[2024-06-30T09:01:47.778+0000] {Load_data.py:240} INFO - Fact_Sales loaded successfully
[2024-06-30T09:01:47.778+0000] {northwindDWH.py:54} INFO - Data delivery done
[2024-06-30T09:01:47.778+0000] {python.py:237} INFO - Done. Returned value was: None
[2024-06-30T09:01:47.779+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-30T09:01:47.787+0000] {taskinstance.py:1206} INFO - Marking task as SUCCESS. dag_id=northwindDWH, task_id=data_delivery, run_id=manual__2024-06-30T08:53:53.031867+00:00, execution_date=20240630T085353, start_date=20240630T090147, end_date=20240630T090147
[2024-06-30T09:01:47.809+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2024-06-30T09:01:47.824+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-06-30T09:01:47.825+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
